base/bucket_gocb.go

```diff
@@ -1795,14 +1795,22 @@ func (bucket *CouchbaseBucketGoCB) Incr(k string, amt, def uint64, exp uint32) (
 	// retry handling, so doesn't need redundant retry handling here.
 	if amt == 0 {
 		var result uint64
-		_, err := bucket.Get(k, &result)
+		cas, err := bucket.Get(k, &result)
 		if bucket.IsKeyNotFoundError(err) {
-			// Return Incr value as zero
-			return uint64(0), nil
+			if def != 0 {
+				// try a cas-safe write to initialize the counter to def
+				_, err := bucket.WriteCas(k, 0, 0, cas, def, sgbucket.Raw)
+				if err != nil {
+					return 0, err
```

Comment:
If another caller initializes the counter while we're attempting to write, we'll get a CAS error here.  I'm not sure it makes sense to return CAS error to the caller in this case - it might be preferable to retry the bucket.Get.  This also applies to transient errors.  This suggests that the amt==0 case should use the same retry handling as the amt!=0 case (below).